pkg/controller/acmechallenges/sync.go

```diff
@@ -131,6 +133,26 @@ func (c *Controller) Sync(ctx context.Context, ch *cmapi.Challenge) (err error)
 		return nil
 	}
 
+	// check for CAA records.
+	// CAA records are static, so we don't have to present anything
+	// before we check for them.
+
+	// Find out which identity the ACME server says it will use.
+	dir, err := cl.Discover(ctx)
+	if err != nil {
+		return err
+	}
+	// TODO(dmo): figure out if missing CAA identity in directory
+	// means no CAA check is performed by ACME server or if any valid
+	// CAA would stop issuance (strongly suspect the former)
```

Comment:
+1 I feel like the former is probably the case here.

@dmo do you have any insight?

Reply:
There's no absolute guarantee that if a CA doesn't include a `caaIdentities` field in the directory metadata that it won't check CAA. For instance, Let's Encrypt was checking CAA [prior to the field being specified or included](https://github.com/letsencrypt/boulder/commit/17922a6d2df0ebdef5ded54fea70ad8f5503ecf9).

That said, if the CA doesn't include any `caaIdentities` there won't be a way for an ACME client to programmatically check whether issuance will pass a CAA check or not. I think for your purposes it would be correct to assume there's no CAA since you can't do much else besides hardcode server URLs -> CAA identities as they arise (ick).